framework/src/bundle/CoreBundleContext.cpp

```diff
@@ -169,20 +169,32 @@ void CoreBundleContext::Uninit1()
   listeners.Clear();
   resolver.Clear();
 
+  // Do not hold the bundleThreads lock while calling
+  // BundleThread::Quit() because this will join the bundle
+  // thread which may need to acquire the bundleThreads lock
+  // itself.
+  //
+  // At this point, all bundles are stopped and all
+  // bundle context instances invalidated. So no new bundle
+  // threads can be created and it is sufficient to get
+  // the current list once and not check for new bundle
+  // threads again.
+  std::list<std::shared_ptr<BundleThread>> threads;
+  bundleThreads.Lock(), std::swap(threads, bundleThreads.value);
```

Comment:
> Scratch that, at this point we called Stop() on all bundles, which internally either joins the bundle thread or puts it back into the list before returning. So this is okay.

Do you mean that at this point (i.e. in ```CoreBundleContext::Uninit1()```), ```BundleThread::Run()``` might execute lines 94 and 95 in ```BundleThreads.cpp```, however it will _never_ enter into the if statement to modify ```bundleThreads.value``` and ```bundleThreads.zombies```?